bundles/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/internal/discovery/EnvisalinkBridgeDiscovery.java

```diff
@@ -57,43 +59,29 @@ public EnvisalinkBridgeDiscovery(DSCAlarmBridgeDiscovery dscAlarmBridgeDiscovery
     public synchronized void discoverBridge() {
         logger.debug("Starting Envisalink Bridge Discovery.");
 
-        SubnetUtils subnetUtils = null;
-        SubnetInfo subnetInfo = null;
-        long lowIP = 0;
-        long highIP = 0;
-
+        CidrAddress localCidrAddress;
         try {
             InetAddress localHost = InetAddress.getLocalHost();
-            NetworkInterface networkInterface = NetworkInterface.getByInetAddress(localHost);
-            subnetUtils = new SubnetUtils(localHost.getHostAddress() + "/"
-                    + networkInterface.getInterfaceAddresses().get(0).getNetworkPrefixLength());
-            subnetInfo = subnetUtils.getInfo();
-            lowIP = convertIPToNumber(subnetInfo.getLowAddress());
-            highIP = convertIPToNumber(subnetInfo.getHighAddress());
-        } catch (IllegalArgumentException e) {
-            logger.warn("discoverBridge(): Illegal Argument Exception - {}", e.toString());
-            return;
-        } catch (Exception e) {
-            logger.warn("discoverBridge(): Error - Unable to get Subnet Information! {}", e.toString());
+            localCidrAddress = NetUtil.getAllInterfaceAddresses().stream()
+                    .filter(f -> f.getAddress() instanceof Inet4Address && f.getAddress().equals(localHost)).findFirst()
+                    .orElse(null);
+        } catch (UnknownHostException e) {
+            logger.warn("discoverBridge(): UnknownHostException - {}", e.toString());
             return;
         }
 
-        logger.debug("   Local IP Address: {} - {}", subnetInfo.getAddress(),
-                convertIPToNumber(subnetInfo.getAddress()));
-        logger.debug("   Subnet:           {} - {}", subnetInfo.getNetworkAddress(),
-                convertIPToNumber(subnetInfo.getNetworkAddress()));
-        logger.debug("   Network Prefix:   {}", subnetInfo.getCidrSignature().split("/")[1]);
-        logger.debug("   Network Mask:     {}", subnetInfo.getNetmask());
-        logger.debug("   Low IP:           {}", convertNumberToIP(lowIP));
-        logger.debug("   High IP:          {}", convertNumberToIP(highIP));
+        List<InetAddress> addressesToScan = localCidrAddress != null
+                ? NetUtil.getAddressesRangeByCidrAddress(localCidrAddress, 16)
```

Comment:
A reason to use 16 ?

Reply:
It's there just to prevent idiotic scans. I would certainly not expect homes to have a larger network then /16.